service/qos/service_level_controller.cc

```diff
@@ -221,6 +223,25 @@ future<> service_level_controller::update_service_levels_from_distributed_data()
     });
 }
 
+void service_level_controller::maybe_start_legacy_update_loop(service::storage_service& storage_service, service::raft_group0_client& group0_client, lw_shared_ptr<db::config> cfg) {
+    assert(this_shard_id() == global_controller);
+
+    if (storage_service.get_topology_upgrade_state() != service::topology::upgrade_state_type::done || group0_client.in_recovery()) {
+        update_from_distributed_data([cfg] () {
```

Comment:
I undestand update_from_distributed_data() is from the old design, but the code now is not obvoius. maybe_start() calls update(), and doesn't start anything. Ideally it should start the loop as its name prescribes. Can update_from_distributed_data() be inlined into maybe_start()?

Reply:
I've merged those 2 functions into one and renamed `update_from_distributed_data()` to `maybe_start_legacy_update_from_distributed_data()`, so the name better reflects what the method does.